hack/python-sdk/main.go

```diff
@@ -23,26 +23,48 @@ import (
 	"strings"
 
 	"github.com/go-openapi/spec"
+	mxJob "github.com/kubeflow/tf-operator/pkg/apis/mxnet/v1"
+	pytorchJob "github.com/kubeflow/tf-operator/pkg/apis/pytorch/v1"
 	tfjob "github.com/kubeflow/tf-operator/pkg/apis/tensorflow/v1"
+	xgboostJob "github.com/kubeflow/tf-operator/pkg/apis/xgboost/v1"
 	"k8s.io/klog"
 	"k8s.io/kube-openapi/pkg/common"
 )
 
 // Generate OpenAPI spec definitions for TFJob Resource
 func main() {
-	if len(os.Args) <= 1 {
-		klog.Fatal("Supply a version")
+	if len(os.Args) <= 2 {
+		klog.Fatal("Supply a framework and version")
 	}
-	version := os.Args[1]
+	framework := os.Args[1]
+	version := os.Args[2]
 	if !strings.HasPrefix(version, "v") {
 		version = "v" + version
 	}
-	oAPIDefs := tfjob.GetOpenAPIDefinitions(func(name string) spec.Ref {
-		return spec.MustCreateRef("#/definitions/" + common.EscapeJsonPointer(swaggify(name)))
-	})
+	var oAPIDefs map[string]common.OpenAPIDefinition
+
+	switch framework {
+	case "tensorflow":
+		oAPIDefs = tfjob.GetOpenAPIDefinitions(func(name string) spec.Ref {
+			return spec.MustCreateRef("#/definitions/" + common.EscapeJsonPointer(swaggify(name, framework)))
+		})
+	case "pytorch":
+		oAPIDefs = pytorchJob.GetOpenAPIDefinitions(func(name string) spec.Ref {
+			return spec.MustCreateRef("#/definitions/" + common.EscapeJsonPointer(swaggify(name, framework)))
+		})
+	case "mxnet":
+		oAPIDefs = mxJob.GetOpenAPIDefinitions(func(name string) spec.Ref {
+			return spec.MustCreateRef("#/definitions/" + common.EscapeJsonPointer(swaggify(name, framework)))
+		})
+	case "xgboost":
+		oAPIDefs = xgboostJob.GetOpenAPIDefinitions(func(name string) spec.Ref {
+			return spec.MustCreateRef("#/definitions/" + common.EscapeJsonPointer(swaggify(name, framework)))
+		})
+	}
```

Comment:
@kubeflow/wg-training-leads  Do we want to have to have multiple or one Swagger for each framework?

Reply:
You mean compress all swagger into a big one?

Reply:
Yes, I am just thinking what is the best way to locate these swaggers.

Reply:
We need to need to generate each for different framework. Then I mixin into one and use it to generate SDK.